pkg/diagnostics/collectors.go

```diff
@@ -435,7 +437,70 @@ func (c *collectorFactory) pingHostCollector(hostIP string) *Collect {
 					Command: []string{"/bin/sh", "-c"},
 					Args:    argsPing,
 				}},
+			}, Timeout: "20s",
+		},
+	}
+}
+
+// vmsAccessCollector will connect to API server first, then collect vsphere-cloud-controller-manager logs
+// on control plane node
+func (c *collectorFactory) vmsAccessCollector(controlPlaneEndpoint string) *Collect {
```

Comment:
This sounds more like a vsphereCloudControllerCollector right?

Reply:
> This sounds more like a vsphereCloudControllerCollector right?

Right, it's vsphere's.

Reply:
yeah it's a bit different than our standard log collectors as it's actually making use directly of a pod execution and explicitly invoking `kubectl`, instead of using the built-in troubleshoot log collectors. It might be good to extend the comment on this collector explaining why we need to target the control plane nodes and use this specific approach

Reply:
iirc and I might be messing this up, the specific log line that we're looking for in these results only occurs on the vsphere-cloud-controller-manager pod on the CP node; if we collect from multiple nodes, such as the worker nodes, it adds a significant amount of noise to the analysis results which could lead to customer confusion. Is that right @YoyoTT ?

Reply:
> 

Yes, that's right! That's also the reason why we want to get the log of vsphere-cloud-controller-manager pod directly from control plane node instead of using the built-in troubleshoot log collector.